colossalai/inference/tensor_parallel/engine.py

```diff
@@ -213,11 +215,14 @@ def _shard_model_by(self, shardformer: ShardFormer, model: nn.Module) -> None:
         ), "Discrepancy between the tp size of TPInferEngine and the tp size of shard config"
         model_name = model.__class__.__name__
         assert model_name in self.supported_models, f"Unsupported model cls {model_name} for TP inference."
-        policy = get_autopolicy(model, inference_only=True)
-        self.model, _ = shardformer.optimize(model, policy)
 
         if self.shard_config.inference_gptq:
-            self._post_init_gptq_buffer(model)
+            policy = get_autopolicy(model.model, inference_only=True)
```

Comment:
The old version is clearer and simpler. You can `to_optimize_model = model.model is self.shard_config.inference_gptq else model`, then follow the old version